perun-registrar-lib/src/main/java/cz/metacentrum/perun/registrar/impl/MailManagerImpl.java

```diff
@@ -171,7 +182,7 @@ public void deleteMailById(PerunSession sess, ApplicationForm form, Integer id)
 		ApplicationMail mail = getMailById(sess, id);
 
 		int result = jdbc.update("delete from application_mails where id=?", id);
-		if (result == 0) throw new InternalErrorException("Mail notification with id="+id+" doesn't exists!");
+		if (result == 0) throw new ApplicationMailAlreadyRemovedException("Mail notification with id="+id+" doesn't exists!");
```

Comment:
Can you please use specific exceptions in the method definition instead of PerunException ? Thank you.